### ISSUE TITLE
[Bug]: A critical distorting of path after zoom canvas

### ISSUE THREAD

**Issue by vitonsky** [fabricjs/fabric.js#8399]
### CheckList

- [X] I agree to follow this project's Code of Conduct
- [X] I have read and followed the [Contributing Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md)
- [X] I have read and followed the [Issue Tracker Guide](https://github.com/fabricjs/fabric.js/blob/master/CONTRIBUTING.md#%EF%B8%8F-issue-tracker)
- [X] I have searched and referenced existing issues and discussions
- [X] I am filing a **BUG** report.
- [X] I have managed to reproduce the bug **after** upgrading to the **latest** version
- [X] I have created an accurate and minimal reproduction

### Version

5.2.1

### In What environments are you experiencing the problem?

Chrome

### Link To Reproduction

https://codesandbox.io/s/delicate-surf-2uq4wr

### Steps To Reproduce

1. Open https://codesandbox.io/s/delicate-surf-2uq4wr
3. Try to change zoom and toggle objectCaching

### Expected Behavior

Figure must be circle. Any zoom changes must not affect on shape of figure

### Actual Behavior

You see a critically distorted figure, it's not a circle.

If you set zoom = 1 or will disable cache, you will see a correct shape.

**Comment by vitonsky** [comment]
Screenshot

<img width="616" src="https://user-images.githubusercontent.com/86191922/198679807-853ae285-c305-4771-b710-4f4ce66dc272.png">

**Comment by ShaMan123** [comment]
I think fabric isn't fully there in term of the arc command.
But I am guessing.
Using Circle as a workaround answers your needs?

**Comment by vitonsky** [comment]
@ShaMan123 no, it works that not only for circle, but for all shapes with rounded border, for example for rhombus

**Comment by ShaMan123** [comment]
Are you talking about stroke bounding box? #8344 ? #8040 ?

**Comment by vitonsky** [comment]
@ShaMan123 yes, it looks like. Is your PR will fix this problem?

**Comment by ShaMan123** [comment]
Should
Polyline/Polygon are fixed
Path still needs looking into
Contributions are welcome

**Comment by vitonsky** [comment]
Why this problem occur only when cache enabled?

**Comment by ShaMan123** [comment]
It isn't really.
As you can see the bbox is wrong when you select the object.
Since caching depends on bbox calculation you see the severity of the bug there as opposed to the selection box which doesn't affect the object's rendering so you may consider that as less acute.
